vignettes/ees-api-workflow.Rmd

```diff
@@ -0,0 +1,282 @@
+connect to the explore education statistics (EES) API to collect data with the help of
+`eesyapi` in R.
+
+As an overview, the stages you will likely need to follow are:
+
+- Find the data set ID
+- Query the data set meta data to identify filter and indicator IDs
+- Design a query using the relevant IDs
+- Post the query to the API
+- Perform some post processing of the data
+
+## Finding a data set ID
+
+Only some data sets on EES are available via the EES API. This is to maintain 
+minimum standards around the API data catalogue.
+
+Assuming you know the data set within a publication that you would like to query, 
+you will first need to identify the data set ID code. There are two methods to do 
+this, either a) via the data set's page on the EES website or b) by querying the 
+API itself.
+
+### Identifying a data set ID via the API
+
+Finding a data set ID via the API is a two step process. First, you'll need to find the parent publication ID and second, you'll need to search the data catalogue within that publication for the data set.
+
+Querying the publication list:
+
+```
+eesyapi::get_publications()
+```
+
+This returns a list of all publications by id and title (along with slug, summary and date last published). 
+From this list, you can find the title of the publication that you're interested in and pick out the 
+associated publication ID.
+
+For example from the test list, we can pick out **"Attendance API data"**, which has an 
+id of **"b6d9ed96-be68-4791-abc3-08dcaba68c04"**. We can now use this in the function `eesyapi::get_data_catalogue()` to find all the data sets within that publication:
+
+```
+eesyapi::get_data_catalogue("b6d9ed96-be68-4791-abc3-08dcaba68c04")
+```
+
+This returns a data frame containing data set ids and titles (along with a summary, the status and information on what's in the latest version). The data set ids can be used as shown in the following sections to get a given data set's summary and meta data, and to query the data set itself.
+
+### Finding a data set ID on the EES website
+
+As with any other data set, you can find an API data set in the EES data catalogue. If you
+know the name, you can filter the catalogue on this as normal. Not all data sets are accessible via the API, so to filter down to just those that are, click the **"API data sets only"** option as shown in the image below.
+
+![Finding an API data set in the EES data catalogue](ees-data-catalogue-find-api-data-set.png)
+
+Once you've found the data set you need, then click on the title of the data set to be 
+taken to it's details page in the catalogue. On this page (pictured below), you'll find the **API data set quick start** section, which gives the API data set name, **API data set ID** and API data set version. This **API data set ID** is what's needed below to query the data set and its meta data.
+
+![An example of an API data set's details](ees-data-catalogue-details-api-data-set.png)
+
+## Getting the data set meta data
```

Comment:
Is there a step before this where we should add a section on the download option that pulls the full file directly into R?

Reply:
Ah, I did have a title for that, but then removed it with the intention of tagging it on at the end. I was thinking of it being after the querying stuff rather than before as people might just decide to stop there once they've got the full data set that they can manipulate locally.

Reply:
They might, but I was thinking it'd work well as we could put clear caveats about how it'll break and just give all data, so to get specific data and have a reliable query read on into the querying section?

Reply:
It's at the end at the moment, I'll bump it up earlier and we can see what they make of it as they're working through.